src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] items;
+    private int size;
+
+    public ArrayList() {
+        items = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        growCheck(size);
+        items[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        rangeCheckForAdd(index);
+        growCheck(size);
+        System.arraycopy(items, index,
+                items, index + 1,
+                size - index);
+        items[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        growCheck(this.size + list.size());
+        int modCount = size;
+        for (int i = 0; i < list.size(); i++) {
+            items[modCount] = list.get(i);
+            modCount++;
+        }
+        size += list.size();
     }
 
     @Override
     public T get(int index) {
-        return null;
+        rangeCheck(index);
+        return (T) items[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        rangeCheck(index);
+        items[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        rangeCheck(index);
+        T result = (T) items[index];
+        if (index == size - 1) {
+            items[index] = null;
+        } else {
+            System.arraycopy(items, index + 1,
+                    items, index,
+                    size - index);
+        }
+        size--;
+        return result;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if (element == get(i) || (element != null
+                    && element.equals(get(i)))) {
+                return remove(i);
+            }
+        }
+        throw new NoSuchElementException("No such element on this list");
```

Comment:
Let's make your exception message more informative. Please, add specific element in this exceptions message (here and elsewhere exceptions). For example:  throw new NoSuchElementException("No such " + element + " on this list");

Reply:
Ok!

Reply:
but i m going to use String.format